java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -788,6 +788,18 @@ private void buildTryStatement(TryStatementTree tryStatementTree) {
     }
   }
 
+  protected void linkToCatchBlocks(Block block, List<Block> catches) {
+    if (block.exitBlock != null && block.exitBlock.isFinallyBlock) {
+      for (Block catchBlock : catches) {
```

Comment:
Matter of style, but I would choose the block with the if and have only one loop for to add successors.